script to manually create a phylo run workflow

ttung opened the pull request:
### Description
In the future, this will be created by a web endpoint, but to build out the rest of the backend, I need a script that does this for now.

Depends on #236

#### Issue
[ch126782](https://app.clubhouse.io/genepi/story/126782/workflow-to-pull-the-data-from-aspen-api-endpoint-and-launch-nextstrain-build)

### Test plan
created a workflow, manual inspection seems to suggest that it was reasonable.

shanaxel42 commented on .hgtags:
```diff
@@ -0,0 +1,6 @@
+c0cda8b0ffb4a4c71feabfdbf01c4aeb2af27157 t1
+0f598f1ff40c91da04278989f1e4c68d00bc87e0 t1
```
rebase badness?

shanaxel42 commented on src/py/aspen/cli/db.py:
```diff
@@ -108,3 +128,115 @@ def import_covidhub_project(
+                or_(
+                    Sample.public_identifier.in_(samples),
+                    and_(
+                        expression.true()
```
What's getting checked here?

ttung replied:
A sample is included if one of the following is true:

1. The sample is explicitly requested using `--sample`
2. `--all-group-samples` is specified AND the sample belongs to the group.

shanaxel42 commented on src/py/aspen/cli/db.py:
```diff
@@ -108,3 +128,115 @@ def import_covidhub_project(
+            aligned_gisaid_dump = (
+                session.query(AlignedGisaidDump)
+                .join(AlignedGisaidDump.producing_workflow)
+                .order_by(Workflow.end_datetime.desc())
```
oh chill, so it just get's the most recent one?

ttung replied:
yup.

shanaxel42 reviewed (approved):
Exciting! Just a couple questions

jgadling commented on src/py/aspen/cli/db.py:
```diff
@@ -168,3 +188,115 @@ def import_covidhub_project(
+        workflow.inputs = list(pathogen_genomes)
+        workflow.inputs.append(aligned_gisaid_dump)
+        workflow.template_file_path = (
+            "workflows/setup_nextstrain_run/builds_template.yaml"
```
Where is this file?

ttung replied:
The workflow uses this file + the build args to generate the builds file.  It should be present in the target git refspec.  I have been iterating on the workflow in [this branch](https://github.com/chanzuckerberg/aspen/tree/ttung/setup-nextstrain-run).

ttung replied:
[here](https://github.com/chanzuckerberg/aspen/blob/ttung/setup-nextstrain-run/src/py/workflows/nextstrain_run/builds_template.yaml) in this case.

ttung commented on src/py/aspen/cli/db.py:
```diff
@@ -168,3 +187,150 @@ def import_covidhub_project(
+
+        workflow_id = workflow.workflow_id
+
+    batch_client = boto3.client("batch")
+    # TODO: in an ideal world, some of these constants should be shared with the
+    # terraform scripts.
+    batch_client.submit_job(
+        jobName="nextstrain",
+        jobQueue="aspen-batch",
+        jobDefinition="aspen-batch-job-definition",
+        containerOverrides={
+            "command": [
+                git_refspec,
+                "src/py/workflows/nextstrain_run/setup.sh",
+                str(workflow_id),
+            ],
+            "vcpus": 32,
+            "memory": 370000,
+        }
+    )
```
I added code to launch the batch job here.

Commit:
script to manually create a phylo run workflow

### Description
In the future, this will be created by a web endpoint, but to build out the rest of the backend, I need a script that does this for now.

#### Issue
[ch126782](https://app.clubhouse.io/genepi/story/126782/workflow-to-pull-the-data-from-aspen-api-endpoint-and-launch-nextstrain-build)

### Test plan
created a workflow, manual inspection seems to suggest that it was reasonable.
